File: 852-peak-index-in-a-mountain-array/852-peak-index-in-a-mountain-array.py
class Solution(object):
    def peakIndexInMountainArray(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        start = 0
        end = len(arr)-1
        while(start<= end):
            mid = (start+end)//2
            if (arr[mid]>arr[mid+1]):
                if(arr[mid]>arr[mid-1]):
                    return mid
                else:
                    end = mid-1 
            else:
                start = mid+1
                
                    

File: 852-peak-index-in-a-mountain-array/852-peak-index-in-a-mountain-array.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var peakIndexInMountainArray = function(arr) {
    let start =0;
    let end = arr.length
    while(start<=end){
        let mid = Math.floor((start+end)/2);
        if(arr[mid]>arr[mid+1]){
            if(arr[mid]>arr[mid-1]){
                return mid;
            }
            else{
                end = mid -1;
            }
        }else{
            start = mid+1
        }
    }
};